metabase/metabase
# Auto apply dashboard filters with the same field

Epic: #35147 

### Description

When applying a parameter mapping to a dashcard, this PR will apply the same mapping to dashcards that have the same field if the card doesn't already have a field applied to it.

I apply this logic within the `DashCardParameterMapper`, and uses an MBQL function to check whether the fields are the same. Keep in mind that only the fields that are exactly matching will be applied - if there's a field that has a filter applied to it, it won't be matched.

### How to verify

1. Create a dashboard and add three cards - two that come from the same table (it can even be the same card) and one that is completely unrelated. For example, make two cards from the Orders table, and one from the Reviews table.
2. Edit the dashboard and add a filter, like `Text contains...`. Then, in one of the cards from the Orders table, add a filter.
3. See that the filter is applied to the other card from the Orders table, but not the one from the reviews table.
4. Now try to change the mapping from one of the cards from the orders table - it should not change the other card.


### Checklist
